pkg/storage/stores/indexshipper/uploads/index_set.go

```diff
@@ -21,7 +21,7 @@ type IndexSet interface {
 	Add(idx index.Index)
 	Upload(ctx context.Context) error
 	Cleanup(indexRetainPeriod time.Duration) error
-	ForEach(ctx context.Context, doneChan <-chan struct{}, callback index.ForEachIndexCallback) error
+	ForEach(callback index.ForEachIndexCallback) error
```

Comment:
is it weird that the uploads and downloads packages have different signatures for this function (one with a context, the other without)? It seems like in the latter that context is only necessary for a logger.